src/tests/htif-yield.lua

```diff
@@ -51,14 +128,51 @@ local yields = {
     { mcycle = 520, data = 27, cmd = YIELD_AUTOMATIC, reason = REASON_TX_REPORT},
 }
 
+local function run_machine_with_uarch(machine)
```

Comment:
No need to be done in this PR, but I think this function deserves to be written in C++ and exposed to a C API and Lua API in the future, so it can be faster, speeding up our CI. Doing a round-trip in Lua and C++ for every step is slow. 
Having an API for this I think it might allow us to simplify other tests/scripts doing the same.

@diegonehab Thoughts?